android/src/main/java/cl/json/social/SingleShareIntent.java

```diff
@@ -54,14 +49,24 @@ public void open(ReadableMap options) throws ActivityNotFoundException {
         //  configure default
         super.open(options);
     }
+
     protected void openIntentChooser() throws ActivityNotFoundException {
-        if(this.options.hasKey("forceDialog") && this.options.getBoolean("forceDialog")){
-             Intent chooser = Intent.createChooser(this.getIntent(), this.chooserTitle);
-             chooser.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-             this.reactContext.startActivity(chooser);
-        }else{
-             this.getIntent().setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-             this.reactContext.startActivity(this.getIntent());
+        if (this.options.hasKey("forceDialog") && this.options.getBoolean("forceDialog")) {
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP_MR1) {
```

Comment:
Should use the static isSupported() method . you have on TargetChosenReceiver to avoid propagating build versions everywhere

Reply:
I have updated the code to use isSupported() in place of build versions.